File: src/main/js/components/reports/count_lap_report.js
'use strict';
import React from "react";
import {Table} from "reactstrap";

class CountLapReport extends React.Component {
    constructor(props) {
        super(props);

    }
    render(){
        let tableRows = [];
        this.props.report.data.map(row=>{
            tableRows.push(<tr key={row.sportsman.id}>
                <td style={{width:'50px'}}>{row.position}</td>
                <td>{row.sportsman.firstName + ' ' + row.sportsman.lastName +(row.sportsman.nick?'['+row.sportsman.nick+']':'')}</td>
                <td>{row.count}</td>
                <td>{row.gap}</td>
                <td>{row.rel}</td>
            </tr>);
        })
        return(
            <div>
                <h5 style={{textAlign:'center', marginBottom:'20px'}}>{this.props.report.report.name}</h5>
                <Table bordered striped hover key="table" className="table-sm">
                    <thead>
                    <tr>
                        <th style={{width:'50px'}}>Pos</th>
                        <th>Sportsmen</th>
                        <th>Laps</th>
                        <th>Gap</th>
                        <th>Rel.</th>
                    </tr>
                    </thead>
                    <tbody>
                    {tableRows}
                    </tbody>
                </Table>
            </div>
        );
    }

}

export default CountLapReport;

File: src/main/java/com/vstimemachine/judge/report/CountLapReport.java
package com.vstimemachine.judge.report;

import com.vstimemachine.judge.model.Sportsman;
import lombok.Data;

@Data
public class CountLapReport implements ReportDataInterface {

    public CountLapReport() {
    }

    public CountLapReport(Sportsman sportsman, Integer count) {
        this.sportsman = sportsman;
        this.count = count;
    }

    public CountLapReport(Integer position, Sportsman sportsman, Integer count, Integer gap, Integer rel) {
        this.position = position;
        this.sportsman = sportsman;
        this.count = count;
        this.gap = gap;
        this.rel = rel;
    }


    private Integer position;
    private Sportsman sportsman;
    private Integer count;
    private Integer gap;
    private Integer rel;
}
